### PR TITLE
Add Akeyless provider

### PR DESCRIPTION
**Pull request by renanaAkeyless**
Add Akeyless provider
support getting your secrets from akeyless

### CODE REVIEW

**Comment by renanaAkeyless** [comment]
@knelasevero @moolen 
can you please take a look at my PR 
thanks :)

File: e2e/run.sh [hunk 1]
```diff
@@ -54,6 +54,9 @@ kubectl run --rm \
   --env="GCP_PROJECT_ID=${GCP_PROJECT_ID:-}" \
   --env="AZURE_CLIENT_ID=${AZURE_CLIENT_ID:-}" \
   --env="AZURE_CLIENT_SECRET=${AZURE_CLIENT_SECRET:-}" \
+  --env="AKEYLESS_ACCESS_ID=${AKEYLESS_ACCESS_ID}" \
```
Comment (knelasevero) [inline e2e/run.sh:57]:
```suggestion
  --env="AKEYLESS_ACCESS_ID=${AKEYLESS_ACCESS_ID:-}" \
```

File: pkg/provider/register/register.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package register
 // packages imported here are registered to the controller schema.
 // nolint:revive
 import (
+	_ "github.com/external-secrets/external-secrets/pkg/provider/akeyless"
```
Comment (knelasevero) [inline pkg/provider/register/register.go:20]: I don't think we are going to have a Akeyless dedicated account for this project, at least right now. I think we can left it out here like we don't have alibaba and gilatb.

File: e2e/run.sh (hunk 1 above)
Comment (renanaAkeyless) [reply e2e/run.sh]: done. changed as suggested above

File: pkg/provider/register/register.go (hunk 2 above)
Comment (knelasevero) [reply pkg/provider/register/register.go:20]: Sorry, I was sleepy, I meant in e2e/suite/import.go, you are right :sweat_smile:

**Comment by knelasevero** [comment]
Can you rebase with main branch please?

File: docs/snippets/akeyless-external-secret-json.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    kind: SecretStore
+    name: akeyless-secret-store # Must match SecretStore on the cluster
+
+  target:
+    name: akeyless-secret-to-create-json # Name for the secret to be created on the cluster
+    creationPolicy: Owner
+
+  # each secret name in the KV will be used as the secret key in the SECRET k8s target object
+  dataFrom:
+  - key: "my-app" # Key of the variable on Gitlab
```
Comment (knelasevero) [inline docs/snippets/akeyless-external-secret-json.yaml:18]:
```suggestion
  - key: "my-app" # Key of the variable on Akeyless
```

Comment (knelasevero) [inline docs/snippets/akeyless-external-secret-json.yaml:18]:
```suggestion
  - key: "my-app" # Key of the variable on Akeyless
```

File: docs/snippets/akeyless-external-secret.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    name: akeyless-secret-store # Must match SecretStore on the cluster
+
+  target:
+    name: akeyless-secret-to-create # Name for the secret to be created on the cluster
+    creationPolicy: Owner
+
+  data:
+    - secretKey: secretKey # Key given to the secret to be created on the cluster
+      remoteRef:
+        key: my-app # Key of the variable on Gitlab
```
Comment (knelasevero) [inline docs/snippets/akeyless-external-secret.yaml:19]:
```suggestion
        key: my-app # Key of the variable on Akeyless
```

**Comment by OriMankali** [comment]
When is this expected to be merged?

**Comment by knelasevero** [comment]
I want to manually test the workflow on the docs one time before merging. I did not have the time to do that yet.

**Comment by renanaAkeyless** [comment]
@knelasevero I can help you test it (with credentials and secret-name). you can PM me for details

**Comment by knelasevero** [comment]
@renanaAkeyless thanks a lot for the contribution, I think we are all good with this working like a charm.

Can you also add the docs in https://github.com/external-secrets/external-secrets/blob/main/hack/api-docs/mkdocs.yml?

you can check docs locally with `make docs`, and `make serve-docs`

**Comment by renanaAkeyless** [comment]
@knelasevero thanks :) i have updated the documentation.

**Comment by adyb-fj-cns** [comment]
The CRDs in the helm chart dont appear to be fully updated. The Cluster Secret Store and Secret Store Providers dont look like they have AKeyless listed, however the external secret does

**Comment by moolen** [comment]
@adyb-fj-cns it hasn't been released yet, CRDs will be up to date with the next release.

**Comment by adyb-fj-cns** [comment]
Ah ok. Perhaps that should be noted in the live documentation as it gives the impression it was released, particularly as some of the CRDS have already been updated

**Pull request merged**